plugins/modules/vault_pki_generate_certificate.py

```diff
@@ -0,0 +1,266 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# (c) 2022, Florent David <florent.david@gmail.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = """
+  module: vault_pki_generate_certificate
+  version_added: 2.11.6
+  author:
+    - Florent David <florent.david@gmail.com>
+  short_description: Generates a new set of credentials (private key and certificate) using HashiCorp Vault PKI
+  requirements:
+    - C(hvac) (L(Python library,https://hvac.readthedocs.io/en/stable/overview.html))
+    - For detailed requirements, see R(the collection requirements page,ansible_collections.community.hashi_vault.docsite.user_guide.requirements).
+  description:
+    - Generates a new set of credentials (private key and certificate) based on a role named and depending PKI's mount point.
+  seealso:
+    - name: Hashicorp Vault PKI Secrets Engine API
+      description: API documentation for the Hashicorp Vault PKI secrets engine.
+      link: https://www.vaultproject.io/api/secret/pki#generate-certificate
+    - name: HVAC library reference
+      description: Reference of the HVAC library about PKI engine.
+      link: https://hvac.readthedocs.io/en/stable/usage/secrets_engines/pki.html#generate-certificate
+  extends_documentation_fragment:
+    - community.hashi_vault.connection
+    - community.hashi_vault.auth
+  options:
+    - alt_names:
```

Comment:
The option names should be keys in the dictionary of `options`, not list items, this is why the tests are failing currently, so:
```yaml
  options:
    alt_names:
      description: ...
    next_option:
      description: etc...
```

Reply:
Fixed in commit 3a81a4ba8da0bef037f518023c3e4553e7c66cda